cassiopeia/simulator/Cas9LineageTracingDataSimulator.py

```diff
@@ -0,0 +1,328 @@
+            can specify. If this argument is not None, states will not be
+            pulled from the state distribution.
+        heritable_silencing_rate: Silencing rate for the cassettes, per node,
+            simulating heritable missing data events.
+        stochastic_silencing_rate: Rate at which to randomly drop out cassettes,
+            to simulate dropout due to low sensitivity of assays.
+        random_seed: Random seed to use for deterministic simulations. Note
+            that the random seed gets set during every call to `overlay_data`,
+            thereby producing deterministic simulations every time this function
+            is called.
+
+    Raises:
+        DataSimulatorError if assumptions about the system are broken.
+    """
+
+    def __init__(
+        self,
+        number_of_cassettes: int = 10,
+        size_of_cassette: int = 3,
+        mutation_rate: Union[float, List[float]] = 0.01,
+        state_generating_distribution: Callable[[], float] = lambda: np.random.exponential(
+            1e-5
+        ),
+        number_of_states: int = 100,
+        state_priors: Optional[Dict[int, float]] = None,
+        heritable_silencing_rate: float = 1e-4,
+        stochastic_silencing_rate: float = 1e-2,
+        random_seed: Optional[int] = None,
+    ):
+
+        if number_of_cassettes <= 0 or not isinstance(number_of_cassettes, int):
+            raise DataSimulatorError("Specify a positive number of cassettes.")
+        if size_of_cassette <= 0 or not isinstance(size_of_cassette, int):
+            raise DataSimulatorError(
+                "Specify a positive number of cut-sites" " per cassette."
+            )
+
+        self.size_of_cassette = size_of_cassette
+        self.number_of_cassettes = number_of_cassettes
+
+        if isinstance(mutation_rate, float):
+            if mutation_rate < 0:
+                raise DataSimulatorError(
+                    "Mutation rate needs to be" " non-negative."
+                )
+            number_of_characters = size_of_cassette * number_of_cassettes
+            self.mutation_rate_per_character = [
+                mutation_rate
+            ] * number_of_characters
+        else:
+            if len(mutation_rate) != (
+                self.number_of_cassettes * self.size_of_cassette
+            ):
+                raise DataSimulatorError(
+                    "Length of mutation rate array is not"
+                    " the same as the number of characters."
+                )
+
+            if np.any(np.array(mutation_rate) < 0):
+                raise DataSimulatorError(
+                    "Mutation rate needs to be" " non-negative."
+                )
+            self.mutation_rate_per_character = mutation_rate
+
+        self.mutation_priors = state_priors
+        if self.mutation_priors is None:
+            self.mutation_priors = {}
+            probabilites = [
+                state_generating_distribution() for _ in range(number_of_states)
```

Comment:
This step happens before a random seed is set, so it might produce different results on different runs of the global script? I would probably move this code out of the `__init__` method and into the `overlay_data` method.

Reply:
Ah, this is a great point! I'll move this to the `overlay_data` method.